project/Yu-Rou-Weng/proposal.md

```diff
@@ -0,0 +1,102 @@
+
+2. SU2 is a suite of open-source software tools written in C++ for the numerical solution 
+  of partial differential equations (PDE) and performing PDE-constrained optimization.
+
+When dealing with numerical analysis, the mesh data format differs between Plot3D and SU2. 
+Plot3D uses structured mesh as input, while SU2 uses unstructured mesh for solving fluid dynamics.
+
+The goal of this project is to create a file format converter that can translate Plot3D mesh data 
+into a format compatible with SU2. This converter will function as a standalone executable, facilitating data exchange between different programs. Also, the converter will be integrated to [modmesh](https://github.com/solvcon/modmesh) to serve as a useful plugged-in library for easier automation.
+
+## Prospective Users
+
+1. [modmesh](https://github.com/solvcon/modmesh):  
+     - modmesh is a toolkit for solving Partial Differential Equations (PDE)  
+     - Fluid flow problems in Computational Fluid Dynamics (CFD) are often   
+     described by PDE
+     - modmesh converts geometric information into computer-readable formats like Plot3D.
+     This converter facilitates seamless integration between modmesh and SU2, enabling 
+     accurate and efficient computational fluid dynamics simulations
+
+2. Fluid dynamics professionals or researchers who need to exchange mesh data between Plot3D and SU2 when using different simulation software.
+
+3. In the aerospace and automotive industries, engineers frequently rely on computational fluid dynamics (CFD) simulations. This converter offers a more precise mesh data conversion solution 
+for their simulations.
+
+4. Computer scientists, students, and educators who are working in computational fluid dynamics studies and experiments.
+
+## System Architecture
+
+![image](https://github.com/Yu-Rou-Weng/nsdhw_23au/blob/Yu-Rou-Weng-Proposal-Submission/project/Yu-Rou-Weng/image/ConceptFlowChart.png)
+
+## API Description
+
+1. Fuctions in c++ will include ReadPlot3DData, WriteSU2Data, ConvertPlot3DToSU2
+   - Use Python to read input and write output data
+   - Use C++ to implement the `Data Conversion` library
+
+2. Use Pybind11 to wrap C++ functions for Python
+
+## Engineering Infrastructure
+
+1. Automatic build system: `CMake`
+2. Version control: `Git`
+3. Testing framework: `Pytest`
+4. Documentation: `README`, [`PyDoc`](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring)
+
+## Schedule
+
+* Week 1 (10/30):  
+     - Study domain knwoledges and meanwhile write corresponding prototypes
+     - Prepare a presentaion skeleton  
+* Week 2 (11/6):  
+     - Search and gather suitable Plot3D mesh data format and comprehend structures and characteristic of both Plot3D and SU2  
+     - Start to design the conversion in C++ if 1. is done in advance
+* Week 3 (11/13):  
+     - Implement conversion in C++ & write pybind11 wrapper
+* Week 4 (11/20):  
+     - Implement conversion in C++ & write pybind11 wrapper
```

Comment:
It may be more productive to include CMake building script along with your C++ code development.  Unit tests may also help your develop efficiency.  But the current plan is OK.  You may give the options a try when you start prototyping.